enterprise_access/apps/api/v1/views/provisioning.py

```diff
@@ -52,8 +53,56 @@ class ProvisioningCreateView(PermissionRequiredMixin, generics.CreateAPIView):
     permission_classes = (permissions.IsAuthenticated,)
     permission_required = constants.PROVISIONING_CREATE_PERMISSION
 
+    def _transform_legacy_request_data(self, request_data: dict) -> dict:
+        """
+        Transform legacy request format to support backward compatibility.
+
+        Converts old `subscription_plan` key to new `trial_subscription_plan` and
+        `first_paid_subscription_plan` keys if the old format is detected.
+
+        Args:
+            request_data (dict): The incoming request data
+
+        Returns:
+            dict: Transformed request data with new keys
+        """
+        # If new format is already present, return as-is.
+        if 'trial_subscription_plan' in request_data and 'first_paid_subscription_plan' in request_data:
+            return request_data
+
+        # If old format is present, transform it.
+        if 'subscription_plan' in request_data:
+            logger.warning(
+                'Deprecated request format detected: `subscription_plan` key should be replaced with '
+                '`trial_subscription_plan` and `first_paid_subscription_plan`'
+            )
+
+            subscription_plan = request_data.pop('subscription_plan')
+
+            # Use the subscription_plan data as trial_subscription_plan
+            request_data['trial_subscription_plan'] = subscription_plan
+
+            # Synthesize first_paid_subscription_plan with required fields only
+            request_data['first_paid_subscription_plan'] = {
+                'title': f"{subscription_plan.get('title', 'Subscription')} - First Paid Plan",
+                'product_id': settings.PROVISIONING_PAID_SUBSCRIPTION_PRODUCT_ID,
+                'salesforce_opportunity_line_item': None,
+            }
+
+            logger.info(
+                'Transformed legacy subscription_plan to trial_subscription_plan and '
+                'synthesized first_paid_subscription_plan'
+            )
```

Comment:
nit: might be helpful during rollout to log the contents of the transformed request data.